Source/WebKit/Shared/Cocoa/CoreIPCArray.h

```diff
@@ -32,24 +32,27 @@
 
 namespace WebKit {
 
+class CoreIPCNSCFObject;
+
 class CoreIPCArray {
 public:
-    CoreIPCArray(NSArray *array)
-        : m_array(array)
-    {
-    }
-
+    CoreIPCArray(NSArray *);
     CoreIPCArray(RetainPtr<NSArray> array)
-        : m_array(WTFMove(array))
+        : CoreIPCArray(array.get())
     {
     }
 
-    RetainPtr<id> toID() { return m_array; }
+    RetainPtr<id> toID() const;
 
 private:
     friend struct IPC::ArgumentCoder<CoreIPCArray, void>;
 
-    IPC::CoreIPCRetainPtr<NSArray> m_array;
+    CoreIPCArray(Vector<std::unique_ptr<CoreIPCNSCFObject>>&& array)
+        : m_array(WTFMove(array))
+    {
+    }
+
+    Vector<std::unique_ptr<CoreIPCNSCFObject>> m_array;
```

Comment:
I don't see any reason to have a unique_ptr here.  Just have a Vector<CoreIPCNSCFObject>

Reply:
TBQH it was to fight forward declaration vs actual includes of the header, as CoreIPCNSCFObject.h includes all of the subordinate headers.

I'll try to rework forward declares vs includes to get this simpler.